### PR TITLE
docs(nextjs): update server components doc

### PR DESCRIPTION
**Pull request by danherrera**
Hello! I was going through this doc as a supabase and nextjs beginner and found a few clarifications I think might help other newbies

### CODE REVIEW

File: apps/docs/pages/guides/auth/auth-helpers/nextjs-server-components.mdx [hunk 1]
```diff
@@ -95,7 +95,7 @@ Create a new file at `/utils/supabase-browser.ts` and populate with the followin
 
 ```ts
 import { createBrowserSupabaseClient } from '@supabase/auth-helpers-nextjs'
-import { Database } from '../db_types'
+import { Database } from '../lib/database.types'
```
Comment (danherrera) [inline apps/docs/pages/guides/auth/auth-helpers/nextjs-server-components.mdx:98]: The [CLI doc](https://supabase.com/docs/reference/javascript/typescript-support) recommends generating types into `lib/database.types.ts`, which are referenced to in [other docs](https://supabase.com/docs/guides/auth/auth-helpers/nextjs) that link to this doc, resulting in a slight discrepancy.

File: apps/docs/pages/guides/auth/auth-helpers/nextjs-server-components.mdx [hunk 2]
```diff
@@ -164,6 +166,8 @@ export const config = {
 
 <TabPanel id="ts" label="TypeScript">
 
+Create a new file at `/pages/middleware.ts` and populate with the following:
```
Comment (danherrera) [inline apps/docs/pages/guides/auth/auth-helpers/nextjs-server-components.mdx:169]: I wasn't sure where to place the middleware file, but got it to work by placing it in the _pages_ directory. Is this the intended location for a middleware?

File: apps/docs/pages/guides/auth/auth-helpers/nextjs-server-components.mdx [hunk 3]
```diff
@@ -254,7 +258,7 @@ export default async function RootLayout({ children }) {
 }
 ```
 
-We don't want Next.js to cache this `session` value, so we need to export a `revalidate` value of `0`.
+We don't want Next.js to cache this `session` value, so we need to export a `revalidate` value of `0` in the layout file.
```
Comment (danherrera) [inline apps/docs/pages/guides/auth/auth-helpers/nextjs-server-components.mdx:261]: Although I was able to infer by looking down further into this doc, it wasn't obvious whether this change still pertained to the layout file

File: apps/docs/pages/guides/auth/auth-helpers/nextjs-server-components.mdx (hunk 2 above)
Comment (saltcod) [reply apps/docs/pages/guides/auth/auth-helpers/nextjs-server-components.mdx:169]: Thanks @danherrera — not anymore. As of 12.2 I think, it's recommended to be at the root — on the same level as /pages/

Comment (saltcod) [inline apps/docs/pages/guides/auth/auth-helpers/nextjs-server-components.mdx:169]:
```suggestion
Create a new file at `/middleware.ts` (a sibling of /pages) and populate with the following:
```

**Comment by saltcod** [comment]
Great improvements @danherrera thanks!

**Review by saltcod — APPROVED** [review]

**Pull request merged**